tests/python/pants_test/engine/test_isolated_process.py

```diff
@@ -308,7 +308,7 @@ def test_integration_concat_with_snapshots_stdout(self):
 
     cat_exe_req = CatExecutionRequest(
       ShellCat(BinaryLocation('/bin/cat')),
-      PathGlobs(include=['f*']),
+      PathGlobs(['f*']),
```

Comment:
I hope not. It was really difficult to develop the script in https://github.com/pantsbuild/pants/pull/9029 in large part because we allow `exclude` to be so flexible, such as supporting `exclude=[['report.py']]`. 

For only two extra characters of `[]`, I think it's worth the consistency to always require a `typing.Iterable`.

Reply:
Ok, I actually agree! That being said I would really love to brainstorm sometime soon about an intelligent BUILD file rewriter. Realistically I think it could be the work of an internship (with a strong candidate and *lots* of support) or potentially a cross-team effort -- can I put some time on your calendar to discuss this perhaps?

And separately, to be clear, does this mean there will be some run-time check that `include` is not a string? Where will that exception occur + will it point to the affected target?

Reply:
> That being said I would really love to brainstorm sometime soon about an intelligent BUILD file rewriter

Toolchain has been working on a rewrite of `buildgen` that I think you'll love. It's not ready quite yet, but we're looking at steps to open source it!

(Part of the reason I'm excited about deprecating `globs` et al. is that it simplifies the grammar for BUILD files so makes it easier to supercharge buildgen even more.)

----

> And separately, to be clear, does this mean there will be some run-time check that include is not a string? Where will that exception occur + will it point to the affected target?

```
diff --git a/src/python/pants/util/BUILD b/src/python/pants/util/BUILD
index ac859dda8..3f0d527a0 100644
--- a/src/python/pants/util/BUILD
+++ b/src/python/pants/util/BUILD
@@ -142,7 +142,7 @@ python_library(
 
 python_library(
   name = 'strutil',
-  sources = ['strutil.py'],
+  sources = ['strutil.py', 1],
   tags = {'type_checked'},
 )

```

```
ValueError: Expected either a glob or list of literal sources. Got: ['strutil.py', 1]
```